plugins/lute-thai/lute_thai_parser/parser.py

```diff
@@ -0,0 +1,69 @@
+"""
+Parsing using pythainlp
+
+Includes classes:
+
+- ThaiParser
+
+"""
+
+import re
+import os
+import pythainlp
+
+from typing import List
+
+from lute.parse.base import ParsedToken, AbstractParser
+
+
+class ThaiParser(AbstractParser):
+    """
+    A parser for Thai that uses the pythainlp library for text segmentation.
+
+    The user can add some exceptions to the "parsing_exceptions.txt"
+    data file.
+    """
+
+    @classmethod
+    def name(cls):
+        return "Lute Thai"
+
+    @classmethod
+    def uses_data_directory(cls):
+        "Uses the data_directory (defined in the AbstractParser)."
+        return False
+
+    # @classmethod
+    # def init_data_directory(cls):
+    #     "Set up necessary files."
+    #     pass
+
+    def get_parsed_tokens(self, text: str, language) -> List[ParsedToken]:
+        """
+        Returns ParsedToken array for given language.
+        """
+        text = text.replace("\r\n", "\n")
+
+        words = pythainlp.word_tokenize(text)
+        tokens = []
+        pattern = f"[{language.word_characters}]"
+        for word in words:
+            is_word_char = re.match(pattern, word) is not None
+            is_end_of_sentence = word in language.regexp_split_sentences
+            if is_end_of_sentence:
+                is_word_char = False
```

Comment:
Just wanted to leave a comment about this logic here,

In Thai the "period" punctuation is `ฯ` and in my tests, the parser was showing it as a word and I figured it didn't make sense for punctuation to be words so that is why I have `is_word_char` set to `False` if the text is a sentence delimiter

Reply:
Right, sometimes there maybe funny cases like this where you have to hardcode it.